hazelcast/src/main/java/com/hazelcast/internal/blackbox/sensorpacks/OperatingSystemSensorPack.java

```diff
@@ -0,0 +1,108 @@
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.OperatingSystemMXBean;
+import java.lang.reflect.Method;
+
+import static com.hazelcast.util.Preconditions.checkNotNull;
+
+/**
+ * A Sensor pack for exposing {@link java.lang.management.OperatingSystemMXBean} sensors.
+ */
+public final class OperatingSystemSensorPack {
+
+    private static final double PERCENTAGE_MULTIPLIER = 100d;
+
+    private OperatingSystemSensorPack() {
+    }
+
+    /**
+     * Registers all the sensors in this sensor pack.
+     *
+     * @param blackbox the Blackbox the sensors are registered on.
+     */
+    public static void register(Blackbox blackbox) {
+        checkNotNull(blackbox, "blackbox");
+
+        OperatingSystemMXBean mxBean = ManagementFactory.getOperatingSystemMXBean();
+
+        registerMethod(blackbox, mxBean, "getCommittedVirtualMemorySize", "os.committedVirtualMemorySize");
+        registerMethod(blackbox, mxBean, "getFreePhysicalMemorySize", "os.freePhysicalMemorySize");
+        registerMethod(blackbox, mxBean, "getFreeSwapSpaceSize", "os.freeSwapSpaceSize");
+        registerMethod(blackbox, mxBean, "getProcessCpuTime", "os.processCpuTime");
+        registerMethod(blackbox, mxBean, "getTotalPhysicalMemorySize", "os.totalPhysicalMemorySize");
+        registerMethod(blackbox, mxBean, "getTotalSwapSpaceSize", "os.totalSwapSpaceSize");
+        registerMethod(blackbox, mxBean, "getMaxFileDescriptorCount", "os.maxFileDescriptorCount");
+        registerMethod(blackbox, mxBean, "getOpenFileDescriptorCount", "os.openFileDescriptorCount");
+        registerMethod(blackbox, mxBean, "getProcessCpuLoad", "os.processCpuLoad");
+        registerMethod(blackbox, mxBean, "getSystemCpuLoad", "os.systemCpuLoad");
+
+        blackbox.register(mxBean, "os.systemLoadAverage",
+                new DoubleSensorInput<OperatingSystemMXBean>() {
+                    @Override
+                    public double get(OperatingSystemMXBean bean) {
+                        return PERCENTAGE_MULTIPLIER * bean.getSystemLoadAverage();
+                    }
+                }
+        );
+    }
+
+
+    // This method doesn't depend on the OperatingSystemMXBean so it can be tested. Due to not knowing
+    // the exact OperatingSystemMXBean class it is very difficult to get this class tested.
+    static void registerMethod(Blackbox blackBox, Object osBean, String methodName, String parameter) {
+        final Method method = getMethod(osBean, methodName);
+
+        if (method == null) {
+            return;
+        }
+
+        if (long.class.equals(method.getReturnType())) {
+            blackBox.register(osBean, parameter,
+                    new LongSensorInput() {
+                        @Override
+                        public long get(Object bean) throws Exception {
+                            return (Long) method.invoke(bean);
+                        }
+                    });
+        } else {
```

Comment:
I'd be defensive here and use `else if (double.class.equals(method.getReturnedType()))`;